base/reflection.jl

```diff
@@ -1500,6 +1500,34 @@ function code_ircode_by_type(
     return asts
 end
 
+
+"""
+    return_types(f::Function, types::DataType=default_tt(f); world::UInt=get_world_counter(), interp::NativeInterpreter=Core.Compiler.NativeInterpreter(world))
+
+Return a list of possible return types for a given function `f` and argument types `types`.
+
+# Example
+```julia
+julia> Base.return_types(sum, Tuple{Vector{Int}})
+1-element Vector{Any}:
+ Int64
+
+julia> Base.return_types(+, Tuple{Any, Any})
+187-element Vector{Any}:
+ Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}
+ Union{Float16, Float32, Float64}
+ BigInt
+ BigInt
+ BigFloat
+ BigFloat
+ BigFloat
+ Any
```

Comment:
```suggestion
julia> methods(sum, (Union{Vector{Int},UnitRange{Int}},))
# 2 methods for generic function "sum" from Base:
 [1] sum(r::AbstractRange{<:Real})
     @ range.jl:1396
 [2] sum(a::AbstractArray; dims, kw...)
     @ reducedim.jl:996
     
julia> Base.return_types(sum, (Union{Vector{Int},UnitRange{Int}},))
2-element Vector{Any}:
 Int64 # the result of inference on sum(r::AbstractRange{<:Real})
 Int64 # the result of inference on sum(a::AbstractArray; dims, kw...)
```
Probably this is more instructive?

Reply:
Agreed, I was not clear of the symmetries with `methods` until I filed this PR